### DISCUSSION TITLE
Help with specifying EoL .net core SDK version in toolset-1804.json

### DISCUSSION THREAD

**Discussion by whiper1** [actions/virtual-environments#4113] [Image configuration]
Hi,

I'm trying to generate a image that includes dotnet Core SDK 2.2.100, [link](https://github.com/dotnet/core/blob/main/release-notes/2.2/2.2.0/2.2.0.md?WT.mc_id=dotnet-35129-website)

How can I modify this part in toolset-1804.json to specifically install this version?

`"dotnet": {
        "aptPackages": [
            "dotnet-sdk-2.2",
            "dotnet-sdk-5.0"
        ],
        "versions": [
            "2.2",
            "5.0"
        ]
    }`

Now it will install the latest version 2.2.207 but for legacy reasons I need to install 2.2.100.

Thank you in advance.

**Comment by MaksimZhukov** [comment]
Hello @whiper1!
To install only the `2.2.100` version of the .NET Core SDK `2.2`, you can simply add it to the `sortedSdks` array in the [dotnetcore-sdk.sh](https://github.com/actions/virtual-environments/blob/main/images/linux/scripts/installers/dotnetcore-sdk.sh#L39) script without updating the `toolset-1804.json` as follows:
```
sortedSdks=$(echo ${sdks[@]} | tr ' ' '\n' | grep -v preview | grep -v rc | grep -v display | cut -d\" -f2 | sort -r | uniq -w 5)
sortedSdks+=("2.2.100")
```

To install all available `2.2.x` versions, you need to apply the changes in `toolset-1804.json` that you mentioned above and update the [dotnetcore-sdk.sh](https://github.com/actions/virtual-environments/blob/main/images/linux/scripts/installers/dotnetcore-sdk.sh#L39) script:
```
sortedSdks=$(echo ${sdks[@]} | tr ' ' '\n' | grep -v preview | grep -v rc | grep -v display | cut -d\" -f2 | sort -u -r)
```

**Reply by whiper1** [reply]
Thank you Maksim!
